src/Microsoft.ML.LightGBM/LightGbmArguments.cs

```diff
@@ -24,12 +24,13 @@
 
 namespace Microsoft.ML.LightGBM
 {
-    public delegate void SignatureLightGBMBooster();
+    internal delegate void SignatureLightGBMBooster();
 
     [TlcModule.ComponentKind("BoosterParameterFunction")]
     public interface ISupportBoosterParameterFactory : IComponentFactory<IBoosterParameter>
```

Comment:
>public [](start = 4, length = 6)

Can you make this interface internal?

Reply:
I tried making it internal but it requires more changes than I thought. So, I am not doing it in this PR. I opened an issue #2559

---
In reply to: [256657661](https://github.com/dotnet/machinelearning/pull/2476#discussion_r256657661) [](ancestors = 256657661)